lib/elixir/lib/port.ex

```diff
@@ -22,6 +22,40 @@ defmodule Port do
   end
 
   @doc """
+  Associates the name with a port identifier. `name`, which must
+  be an atom, can be used instead of the port identifier with the
+  `Kernel.send/2` function.
+
+  `Port.register/2` will fail with `ArgumentError` if the port supplied
+  is closed, or if the name is already registered (check with `whereis/1`).
+  """
+  @spec register(port, atom) :: true
+  def register(port, name) when not name in [nil, false, true] do
```

Comment:
Should we check if the first argument is actually a port? This patch has me a bit torn because the stuff in Process will mostly work with ports too.